webSockets/factory/factory.go

```diff
@@ -15,78 +13,55 @@ import (
 	WithAcknowledge          bool
 }
 
-type webSocketsDriverFactory struct {
-	webSocketConfig          outportData.WebSocketConfig
-	marshaller               marshal.Marshalizer
-	uint64ByteSliceConverter connection.Uint64ByteSliceConverter
-	log                      core.Logger
-	withAcknowledge          bool
-}
-
-// NewWebSocketsDriverFactory will return a new instance of outportDriverWebSocketSenderFactory
-func NewWebSocketsDriverFactory(args ArgsWebSocketsDriverFactory) (*webSocketsDriverFactory, error) {
-	if check.IfNil(args.Marshaller) {
-		return nil, outportData.ErrNilMarshaller
-	}
-	if check.IfNil(args.Uint64ByteSliceConverter) {
-		return nil, outportData.ErrNilUint64ByteSliceConverter
-	}
-	if check.IfNil(args.Log) {
-		return nil, outportData.ErrNilLogger
-	}
-	return &webSocketsDriverFactory{
-		webSocketConfig:          args.WebSocketConfig,
-		marshaller:               args.Marshaller,
-		uint64ByteSliceConverter: args.Uint64ByteSliceConverter,
-		withAcknowledge:          args.WithAcknowledge,
-		log:                      args.Log,
-	}, nil
-}
-
-// Create will handle the creation of all the components needed to create an outport driver that sends data over
-// web socket and return it afterwards
-func (o *webSocketsDriverFactory) Create() (webSockets.Driver, error) {
+// NewWebSocketsDriver will handle the creation of all the components needed to create an outport driver that sends data over
+func NewWebSocketsDriver(args ArgsWebSocketsDriverFactory) (webSockets.Driver, error) {
 	var host webSockets.HostWebSockets
 	var err error
-	if o.webSocketConfig.IsServer {
-		host, err = o.createWebSocketsServer()
+	if args.WebSocketConfig.IsServer {
+		host, err = createWebSocketsServer(args)
 	} else {
-		host, err = client.NewWebSocketsClient(client.ArgsWebSocketsClient{
-			RetryDurationInSeconds:   o.webSocketConfig.RetryDurationInSec,
-			WithAcknowledge:          o.withAcknowledge,
-			URL:                      o.webSocketConfig.URL,
-			Uint64ByteSliceConverter: o.uint64ByteSliceConverter,
-			Log:                      o.log,
-			BlockingAckOnError:       false,
-		})
+		host, err = createWebSocketsClient(args)
 	}
 
 	if err != nil {
 		return nil, err
 	}
 
+	host.Start()
+
 	return webSockets.NewWebsocketsDriver(
 		webSockets.ArgsWebSocketsDriver{
-			Marshaller:               o.marshaller,
+			Marshaller:               args.Marshaller,
 			WebsocketSender:          host,
-			Uint64ByteSliceConverter: o.uint64ByteSliceConverter,
-			Log:                      o.log,
+			Uint64ByteSliceConverter: args.Uint64ByteSliceConverter,
+			Log:                      args.Log,
 		},
 	)
 }
 
-func (o *webSocketsDriverFactory) createWebSocketsServer() (webSockets.HostWebSockets, error) {
+func createWebSocketsClient(args ArgsWebSocketsDriverFactory) (webSockets.HostWebSockets, error) {
+	return client.NewWebSocketsClient(client.ArgsWebSocketsClient{
+		RetryDurationInSeconds:   args.WebSocketConfig.RetryDurationInSec,
```

Comment:
Future work for next PR: merge the ArgsWebSocketsClient and ArgsWebSocketsServer as they look the same and remove the duplicated arguments build